region/lowernorfair/west/Screw Attack Room.json

```diff
@@ -1270,33 +2072,124 @@
           ]
         }
       ],
-      "note": "Diagonally shinespark towards the middle door.",
-      "devNote": [
-        "Obstacle C is indicating that Samus has a shinecharge ready to be used.",
-        "Useful only if you cannot reach the door at 2."
+      "note": "Diagonally shinespark towards the middle door."
+    },
+    {
+      "link": [4, 2],
+      "name": "Start Shinecharged, Leave With Spark (HiJump, Bottom Position)",
+      "startsWithShineCharge": true,
+      "requires": [
+        "HiJump",
+        "canShinechargeMovementComplex",
+        {"or": [
+          "ScrewAttack",
+          {"obstaclesCleared": ["B"]}
+        ]},
+        {"shineChargeFrames": 80},
+        {"heatFrames": 105},
+        {"shinespark": {"frames": 2, "excessFrames": 0}}
+      ],
+      "exitCondition": {
+        "leaveWithSpark": {
+          "position": "bottom"
+        }
+      },
+      "unlocksDoors": [
+        {
+          "types": ["super"],
+          "requires": []
+        },
+        {
+          "types": ["missiles", "powerbomb"],
+          "requires": [
+            "never"
+          ]
+        }
+      ]
+    },
+    {
+      "link": [4, 2],
+      "name": "Start Shinecharged, Leave With Spark (HiJump, Top Position)",
+      "startsWithShineCharge": true,
+      "requires": [
+        "HiJump",
+        "canShinechargeMovementComplex",
+        {"or": [
+          "ScrewAttack",
+          {"obstaclesCleared": ["B"]}
+        ]},
+        {"shineChargeFrames": 95},
+        {"heatFrames": 120},
+        {"shinespark": {"frames": 2, "excessFrames": 0}}
+      ],
+      "exitCondition": {
+        "leaveWithSpark": {
+          "position": "top"
+        }
+      },
+      "unlocksDoors": [
+        {
+          "types": ["super"],
+          "requires": []
+        },
+        {
+          "types": ["missiles", "powerbomb"],
+          "requires": [
+            "never"
+          ]
+        }
+      ]
+    },
+    {
+      "link": [4, 2],
+      "name": "Start Shinecharged, Leave Shinecharged (HiJump)",
+      "startsWithShineCharge": true,
+      "requires": [
+        "HiJump",
+        "canShinechargeMovementComplex",
+        {"or": [
+          "ScrewAttack",
+          {"obstaclesCleared": ["B"]}
```

Comment:
Is it possible for B to be broken?

Reply:
The idea was in case you fell through with temporary blue from above. I guess there's not quite enough shinecharge frames remaining, but it's only 5 short so maybe it's conceivable if something changes here later. Also someone might want to add an option someday for longer shinecharge timer (e.g. 240 frames); there would be lots of stuff missing in that case, but I figure no harm in having stuff be more complete where we can